### DISCUSSION TITLE
I'm an advanced user now :)

### DISCUSSION THREAD

**Discussion by u3erprutser** [uBlockOrigin/uAssets#21498] [Q&A]
I've been using ublock origin in the default settings for a few years now, and it's been great  :)
In an attempt to block an annoying popup element, and to learn a bit more, I just made myself an advanced user.
But, unfortunately, I can't figure it out, and could use some help.

The offending website is `https://www.buienradar.nl`

It has a GDPR cookie banner that is being removed with the "I still don't care about cookies" extension.
But then there is a second popup question that I always have to answer :(
And I'm hoping uBlock can help me with that one.
I've tried the element picker, and it does make the popup go away, but then the site doesn't load properly. It stays greyed out.
I'm hoping to find some help here, because this is driving me crazy :)

**Comment by krystian3w** [comment]
If you don't know JavaScript, it's really difficult to close this pop-up about consent to geolocation.

Theoretically, for people who can't figure it out without trusted scriptlets, this will work: 

- in the settings that I am an advanced user, we restore my filters pane as a trusted space/place at "key": 

```yml
trustedListPrefixes ublock- user-
```

Then works one of scriptlets:

```adb
! Allow geolocation and move protection to browser settings only
buienradar.nl##+js(trusted-set-local-storage-item, buienradar.positionconsent, '{"consent":true,"timestamp":"2023-12-20T09:21:56.723Z"}')
! Better may works trusted-click-element if pop-up are stable and webmaster protect SSH to login at server
```

```adb
! Close pop-up with disallow geolocation at webpage checker
buienradar.nl##+js(trusted-set-local-storage-item, buienradar.positionconsent, '{"consent":false,"timestamp":"2023-12-20T09:24:30.277Z"}')
! Pop-up will back if you click in "found my locaction" in town/city field
! but scriptlet may can block allowing at refresh
! and page create empty value for this localStorage key so don't help ", dontOverwrite, 1" if was copied to more trusted scriptlets
```

---

In the future, I would recommend moving such scriptlets to your own list and allowing them as trusted ones instead of 'user-'. As long someone write these with standard;

- aopr
- aopw
- aeld
- acs (acis)
- set
- aost

Scriptlets with allow geolocation or reject without big breakage (very similar to https://github.com/uBlockOrigin/uBlock-issues/discussions/3034#discussioncomment-7906592).

P.S. my JavaScript knowledge doesn't allow me to convert this to a standard scriptlet(s).

**Reply by krystian3w** [reply]
@Nojuuu Can you check if it's possible to trick this website into allowing or disallowing geolocation so that it doesn't read it from localStorage?

**Comment by krystian3w** [comment]
IMO hide pop-up by only css is bad - break button showed in input form to write town/city manually.

**Comment by u3erprutser** [comment]
I might be an advanced user now, but this all a bit over my head :)

I've playing around on my linux laptop, and it appears to be working now, somehow??? (don't really understand why)
I tried the sync (ff sync) the settings to my Windows computer, and it did not work there??

This is in my filter list. And I believe this is what might be blocking that pop-up
`
||processing-cdn.buienradar.nl/processing/nl/rain/forecast/runs/webm/202312191700/202312191745.png$image
`
After a while, I also got it working on Windows. Not sure why it now works?? Perhaps I should have used the 'purge all caches' function.

fyi: Using the Brave browser with the shield, this pop-up seems to be blocked by default. Perhaps this can be copied to ublock?

note: I just tried the above line on my Windows laptop, and I'm getting confusing results. Now it looks like the pop-up is no longer there with the default setting?? I'm starting to think, someone somewhere did some sort of global update. Anyway, I'm happy with the result :)

note2: I just tried the same setting on another computer, that is not in the same sync chain, and on that Windows pc the popup screen is back again. Could it have anything to do with the .jpg name? I it perhaps possible to use wildcards instead of the full name?

**Reply by krystian3w** [reply]
Sync method works more are network* backup for only one Linux station.

\* - if save to Mozilla Account fails then is only second local backup in other "database".

**Reply by krystian3w** [reply]
I don't know how Brave hides it, so there's no guarantee that there's anything to copy.

**Reply by krystian3w** [reply]
Possible use wildcards or shorter link:

```adb
||processing-cdn.buienradar.nl/processing/nl/rain/forecast/runs/webm/$image
!
||processing-cdn.buienradar.nl/processing/nl/rain/forecast/runs/webm/*.*g$image
```

**Comment by JohnyP36** [comment]
> But then there is a second popup question that I always have to answer :(
And I'm hoping uBlock can help me with that one.
I've tried the element picker, and it does make the popup go away, but then the site doesn't load properly. It stays greyed out.
I'm hoping to find some help here, because this is driving me crazy :)

For me this works `buienradar.nl##+js(trusted-click-element, .onboard-popup-container .onboard-button.button-no)`
*It clicks on "no thanks".*

---

Don't forget to check if 
```
trustedListPrefixes ublock- user-
```
is the case in your "settings that I am an advanced user"
